torch/csrc/jit/script/compiler.cpp

```diff
@@ -223,6 +223,26 @@ struct Environment {
       }
     }
   }
+
+  // for each captured input, create/refer new one in enclosing
+  // scope, and replace references in the block, delete branch block
+  // inputs, and the corresponding field in value_map
+  void deleteIfCapturedInputs(const SourceRange& loc) {
+    // block inputs should be 1:1 to the captured_inputs
+    JIT_ASSERT(b->inputs().size() == captured_inputs.size());
+    for(int i = captured_inputs.size() - 1; i >= 0; -- i) {
+      auto name = captured_inputs[i];
+      auto ident = Ident::create(loc, name);
+      auto parent_frame_var = next->getVar(ident);
+      auto captured_val = b->inputs()[i];
```

Comment:
Can you put explicit types here instead of a bunch of `auto`s? It's harder to tell what's going on at a glance as it is